src/mono/wasm/build/WasmApp.targets

```diff
@@ -207,15 +229,25 @@
       <EmccFlags>$(_DefaultEmccFlags) $(EmccFlags)</EmccFlags>
       <EmccFlags>$(EmccFlags) -s DISABLE_EXCEPTION_CATCHING=0</EmccFlags>
       <EmccFlags Condition="'$(RunAOTCompilation)' == 'true'">$(EmccFlags) -DENABLE_AOT=1 -DDRIVER_GEN=1</EmccFlags>
-      <_EmsdkEnvSourceCommand>source $(EMSDK_PATH)/emsdk_env.sh &gt; /dev/null 2&gt;&amp;1</_EmsdkEnvSourceCommand>
+    </PropertyGroup>
+
+    <PropertyGroup>
+      <_EmsdkEnvSourceCommand Condition="$([MSBuild]::IsOSPlatform('WINDOWS'))">$(EMSDK_PATH)\emsdk_env.bat</_EmsdkEnvSourceCommand>
+      <_EmsdkEnvSourceCommand Condition="!$([MSBuild]::IsOSPlatform('WINDOWS'))">source $(EMSDK_PATH)/emsdk_env.sh &gt; /dev/null 2&gt;&amp;1</_EmsdkEnvSourceCommand>
+
       <_EmccCommand>$(_EmsdkEnvSourceCommand) &amp;&amp; emcc</_EmccCommand>
     </PropertyGroup>
 
     <ReadLinesFromFile File="$(RuntimeEmccVersionFile)">
         <Output TaskParameter="Lines" PropertyName="RuntimeEmccVersion" />
     </ReadLinesFromFile>
 
-    <Exec Command="bash -c '$(_EmccCommand) --version | head -1 > emcc-version.txt'" IgnoreStandardErrorWarningFormat="true" WorkingDirectory="$(_WasmIntermediateOutputPath)" />
+    <Exec Condition="!$([MSBuild]::IsOSPlatform('WINDOWS'))" IgnoreStandardErrorWarningFormat="true" WorkingDirectory="$(_WasmIntermediateOutputPath)"
+          Command="bash -c '$(_EmccCommand) --version | head -1 > emcc-version.txt'" />
```

Comment:
Could you file a tracking issue to determine if shelling out to bash could be removed? It adds an extra dependency that might not be available in all environments.

Reply:
On non-windows platforms, we would it, I believe, to run `emsdk_env.sh`. What platforms might have it missing, and we would be able to install Emscripten?